File: calculateRectangleArea.js
// function calculateArea(length, width) {
//    if (length <= 0 || width <= 0) {
//       throw new Error("Length and width must be positive numbers.");
//    }
//    return length * width;
// }

// const length = 5;
// const width = 3;
// try {
//    const area = calculateArea(length, width);
//    console.log(`The area of the rectangle is: ${area}`);
// } catch (error) {
//    console.error(error.message);
// }

class Rectangle {
   constructor(length, width) {
      if (length <= 0 || width <= 0) {
         throw new Error("Length and width must be positive numbers.");
      }
      this.length = length;
      this.width = width;
   }

   calculateArea() {
      return this.length * this.width;
   }
}

// Example usage:
try {
   const rectangle = new Rectangle(5, 3);
   const area = rectangle.calculateArea();
   console.log(`The area of the rectangle is: ${area}`);
} catch (error) {
   console.error(error.message);
}

File: calculateRectangleArea.py
# def calculate_area(length, width):
#     if length <= 0 or width <= 0:
#         raise ValueError("Length and width must be positive numbers.")
#     return length * width

# length = 5
# width = 3
# try:
#     area = calculate_area(length, width)
#     print(f"The area of the rectangle is: {area}")
# except ValueError as e:
#     print(e)


# class Rectangle:
#     def __init__(self, length, width):
#         if length <= 0 or width <= 0:
#             raise ValueError("Length and width must be positive numbers.")
#         self.length = length
#         self.width = width

#     def calculate_area(self):
#         return self.length * self.width

# try:
#     rectangle = Rectangle(5, 3)
#     area = rectangle.calculate_area()
#     print(f"The area of the rectangle is: {area}")
# except ValueError as e:
#     print(e)

calculate_area = lambda length, width: length * width if length > 0 and width > 0 else ValueError("Length and width must be positive numbers.")

# Example usage:
length = 5
width = 3
try:
    area = calculate_area(length, width)
    print(f"The area of the rectangle is: {area}")
except ValueError as e:
    print(e)
